File: backend/src/main/java/com/ebikko/merchant/model/MerchantSettings.java
package com.ebikko.merchant.model;

import com.ebikko.mandate.model.MandateFrequency;

import javax.persistence.Column;
import javax.persistence.ElementCollection;
import javax.persistence.Embeddable;
import java.util.Set;

@Embeddable
public class MerchantSettings {

    @Column
    private String logoFileLocation;
    @ElementCollection
    private Set<MandateFrequency> selectedFrequencies;
    @ElementCollection
    private Set<String> purposeOfPayments;

    public MerchantSettings() {}

    public String getLogoFileLocation() {
        return logoFileLocation;
    }

    public void setLogoFileLocation(String logoFileLocation) {
        this.logoFileLocation = logoFileLocation;
    }

    public Set<MandateFrequency> getSelectedFrequencies() {
        return selectedFrequencies;
    }

    public void setSelectedFrequencies(Set<MandateFrequency> selectedFrequencies) {
        this.selectedFrequencies = selectedFrequencies;
    }

    public Set<String> getPurposeOfPayments() {
        return purposeOfPayments;
    }

    public void setPurposeOfPayments(Set<String> purposeOfPayments) {
        this.purposeOfPayments = purposeOfPayments;
    }
}

File: frontend/src/app/merchant/model/merchant-settings.ts
import { Serializable } from '../../model/serializable';
import { DisplayEnum } from '../../model/displayEnum';

export class MerchantSettings implements Serializable<MerchantSettings> {

	public logoFileLocation: string;
	public selectedFrequencies: DisplayEnum[]
	public purposeOfPayments: string[];

	deserialize(input) {
		this.logoFileLocation = input.logoFileLocation;
		this.selectedFrequencies = input.selectedFrequencies.map(x => new DisplayEnum().deserialize(x));
		this.purposeOfPayments = input.purposeOfPayments;
		return this;
	}
}